packages/actions/src/internal/decodeRevertReason.js

```diff
@@ -0,0 +1,74 @@
+import { decodeAbiParameters, hexToString } from 'viem'
+
+/**
+ * Decodes basic revert data similar to Anvil's approach
+ * @param {import('@tevm/utils').Hex | string | undefined} data - The revert data
+ * @returns {string} - The decoded reason or 'execution reverted' if not decodable
+ */
+export const decodeRevertReason = (data) => {
+	if (!data || data === '0x') return 'execution reverted'
+	/** @type {import('viem').Hex} */
+	const hexData = typeof data === 'string' && data.startsWith('0x') ? /** @type {import('viem').Hex} */ (data) : /** @type {import('viem').Hex} */ (`0x${data}`)
+
+	// Handle short or malformed data early
+	const cleanHex = hexData.slice(2)
+	if (cleanHex.length === 0 || cleanHex.length % 2 !== 0 || cleanHex.length < 8) {
+		return 'execution reverted'
+	}
+
+	const ERROR_SELECTOR = '0x08c379a0' // Error(string) selector
+	const PANIC_SELECTOR = '0x4e487b71' // Panic(uint256) selector
+
+	// 1. Try Error(string) decoding
+	if (hexData.startsWith(ERROR_SELECTOR)) {
```

Comment:
```suggestion
	if (hexData.toLowercase().startsWith(ERROR_SELECTOR)) {
```
Should be defensive here